pkg/reconcile/reconcilers_test.go

```diff
@@ -307,6 +307,46 @@ func TestPodSecurityContext(t *testing.T) {
 	assert.True(t, same, "podSecurityContext expected at pod level")
 }
 
+func TestVolumes(t *testing.T) {
+	image := "test/reaper:latest"
+	volumes := []corev1.Volume{
+		{
+			Name: "reaper-config",
+		},
+	}
+
+	reaper := newReaperWithCassandraBackend()
+	reaper.Spec.Image = image
+
+	deployment := newDeployment(reaper, "target-datacenter-service")
+	assert.ElementsMatch(t, volumes, deployment.Spec.Template.Spec.Volumes)
+}
+
+func TestInitContainerVolumesMounts(t *testing.T) {
+	image := "test/reaper:latest"
+	volumes := []corev1.Volume{
+		{
+			Name: "reaper-config",
+		},
+	}
+
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "reaper-config",
+			ReadOnly:  true,
+			MountPath: "/etc/reaper",
+		},
+	}
+
+	reaper := newReaperWithCassandraBackend()
+	reaper.Spec.Image = image
+
+	deployment := newDeployment(reaper, "target-datacenter-service")
+
+	assert.ElementsMatch(t, volumes, deployment.Spec.Template.Spec.Volumes)
+	assert.ElementsMatch(t, volumeMounts, deployment.Spec.Template.Spec.Containers[0].VolumeMounts)
```

Comment:
This isn't testing init containers. It looks like the `newDeployment` function needs to be updated to add the volume mount to the `reaper-schema-init` init container.

Reply:
@jsanda Good catch, the init container is where this should be associating the mount paths.  Updated to reflect this.